src/wasm-emscripten.cpp

```diff
@@ -445,8 +453,50 @@ AsmConstWalker fixEmAsmConstsAndReturnWalker(Module& wasm) {
   return walker;
 }
 
+struct EmJsWalker : public PostWalker<EmJsWalker> {
+  Module& wasm;
+  std::vector<Address> segmentOffsets; // segment index => address offset
+
+  std::map<std::string, std::string> codeByName;
+
+  EmJsWalker(Module& _wasm)
+    : wasm(_wasm),
+      segmentOffsets(getSegmentOffsets(wasm)) { }
+
+  void visitFunction(Function* curr) {
+    if (!curr->name.startsWith(EM_JS_PREFIX.str)) {
+      return;
+    }
+    auto funcName = std::string(curr->name.stripPrefix(EM_JS_PREFIX.str));
+    Const* addrConst = curr->body->dynCast<Const>();
+    if (addrConst == nullptr) {
+      addrConst = curr->body->cast<Block>()->list[0]->cast<Const>();
```

Comment:
This is parsing output from the compiler, right? I suppose we'll assert here if we unexpectedly get an emjs function that's not just a const? Maybe we should make a more explicit error check that also fires in release mode.

Reply:
Yeah, seems reasonable.

The idea here was that we should be the only ones generating these __em_js__* functions, so we should know if/when we ever change their format.
But if a user ever does prefix that, we might as well give a saner-sounding "hey no don't".